### PR TITLE
Use ECS container metadata rather than cgroup hack

### PR DESCRIPTION
**Pull request by Sayan-**
For context:
https://clever.atlassian.net/browse/INFRANG-1116
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html

Testing:
Will try this out with `turk-herder` or something prior to merging

### CODE REVIEW

File: cmd/sfncli/ecs.go [hunk 1]
```diff
@@ -1,133 +1,66 @@
...
-	if err != nil {
-		return "", err
+	// wait for the file to exist
+	ticker := time.NewTicker(1 * time.Second)
+	for {
+		<-ticker.C
+		if _, err := os.Stat(filePath); err == nil {
+			break
+		}
 	}
```
Comment (rgarcia) [inline cmd/sfncli/ecs.go]: looks like this waits forever, which doesn't seem ideal...

File: cmd/sfncli/ecs.go [hunk 2]
```diff
@@ -1,133 +1,66 @@
...
-
-	if agentMetadata.Cluster == "" {
-		return "", fmt.Errorf("Could not find ECS cluster for docker container '%s'", dockerID)
-	}
-
-	agentTaskMetadata, err := ecsAgentTaskMetadata()
-	if err != nil {
-		return "", err
+	// wait for the file to exist
+	ticker := time.NewTicker(1 * time.Second)
```
Comment (rgarcia) [inline cmd/sfncli/ecs.go]: defer stopping the ticker

File: cmd/sfncli/ecs.go [hunk 3]
```diff
@@ -1,133 +1,66 @@
...
+		<-ticker.C
+		f, err := os.Open(filePath)
+		if err != nil {
+			return "", err
+		}
+		defer f.Close()
+		b, err := ioutil.ReadAll(f)
+		if err != nil {
+			return "", err
+		}
```
Comment (rgarcia) [inline cmd/sfncli/ecs.go]: can combine these calls into one: https://golang.org/pkg/io/ioutil/#ReadFile

File: cmd/sfncli/ecs.go (hunk 1 above)
Comment (Sayan-) [reply cmd/sfncli/ecs.go]: Hmmm I'm not too sure about that - here's my local flow https://gist.github.com/Sayan-/ffdaac6ec0d9f87cb218da50150362b2

**Comment by Sayan-** [comment]
Verified by:
- building sfncli binary locally
- copying the release into `turk-herder` (https://github.com/Clever/turk-herder/commit/c1517f0e96d4f138a90787d80d75d316c7695690)
- running in dev with another dummy job
- verifying the execution was successful

**Pull request merged**